### ISSUE TITLE
Is there any way to set an initial value and listen to a changed value?

### ISSUE THREAD

**Issue by augustosamame** [jaing/react-native-places-input#12]
I have a form with several TextInputs that have common behaviour. One of this fields is now a PlacesInput. But I'm having trouble trying to replicate the behaviour of a regular TextInput with this component. For example, I'm missing how to change the initial value of the field based on the initial state. Also how to listen to onChangeText to adjust fontSize on the fly in the case of long inputs, keyboardType, placeholderTextColor, etc...
Is there any way this component can be made to extend a regular TextInput ? Just with a displayable list of locations as user enters the value?

My current implementation:

```
<PlacesInput
                googleApiKey={GOOGLE_PLACES_API_KEY}
                language={'en-US'}
                placeHolder={''}
                onSelect={(place) => this.handleSelectedPlace(place, 'home')}
                queryTypes={'(cities)'}
                stylesInput={[(homeTown ? styles.placesField : styles.emptyPlacesField),
                  ( !homeTownFocused && homeTown.length === 0) ? styles.placeholderStyle : '' ]}
                stylesContainer={{
                  position: 'relative',
                  alignSelf: 'stretch',
                  margin: 0,
                  top: 0,
                  left: 0,
                  right: 0,
                  bottom: 0,
                  shadowOpacity: 0,
                  marginBottom: 10,
                }}
                stylesList={{
                  top: 0,
                  borderColor: '#dedede',
                  borderLeftWidth: 1,
                  borderRightWidth: 1,
                  borderBottomWidth: 1,
                  left: -1,
                  right: -1,
                }}
              />
```

**Comment by jaing** [comment]
There is a textInputProps where you can pass text input props. Let me know if that resolve your issue - if not let me know what you are missing

**Comment by augustosamame** [comment]
Thank you for your quick reply. I tried using that prop to set the initial value, but then it can't be changed without overriding onChangeText, which will disable to Google Places Lookup.

textInputProps={{
                  value: this.state.homeTown,
                  onChangeText: (val) => this.setState({ homeTown: val }),
                }}

Like I mentioned, my objective is mostly keeping my (fairly complex) TextInput component and extending it with Google Places functionality. 

I found another library, which I have not tried yet, which offers a programatic call to the Google API when you wish to have control over the UI component:

https://github.com/tolu360/react-native-google-places#using-your-own-custom-uiviews

or any other method by which we can have the logic with our own TextInput.

**Comment by jaing** [comment]
I just published version 1.1.1 with onChangeText prop. It's a function that will return current query and this of a component so you can access state, props etc. 

```
<Component onChangeText={(query, componentScope) => console.log(query, componentScope)} />
```
